engine/access/rpc/backend/backend_scripts.go

```diff
@@ -94,17 +112,55 @@ func (b *backendScripts) executeScriptOnExecutor(
 	script []byte,
 	arguments [][]byte,
 ) ([]byte, error) {
-	// find few execution nodes which have executed the block earlier and provided an execution receipt for it
-	executors, err := executionNodesForBlockID(ctx, blockID, b.executionReceipts, b.state, b.log)
-	if err != nil {
-		return nil, status.Errorf(codes.Internal, "failed to find script executors at blockId %v: %v", blockID.String(), err)
-	}
 	// encode to MD5 as low compute/memory lookup key
 	// CAUTION: cryptographically insecure md5 is used here, but only to de-duplicate logs.
 	// *DO NOT* use this hash for any protocol-related or cryptographic functions.
 	insecureScriptHash := md5.Sum(script) //nolint:gosec
+	// try execution on Archive nodes first
+	archiveResult, err := b.executeScriptOnAvailableArchiveNodes(ctx, blockID, script, arguments, insecureScriptHash)
+	// try execution nodes if the script wasn't executed
+	if b.scriptExecValidation {
+		execNodeResult, errExec := b.executeScriptOnAvailableExecutionNodes(
+			ctx, blockID, script, arguments, insecureScriptHash)
+		if status.Code(err) != codes.NotFound {
+			if bytes.Equal(execNodeResult, archiveResult) && errExec == err {
+				b.logScriptExecutionComparison(blockID, insecureScriptHash,
+					"script execution results on Archive node and EN are equal")
+			} else {
+				b.logScriptExecutionComparison(blockID, insecureScriptHash,
```

Comment:
this might not always be an error, because EN can only execute scripts for last x number of blocks. if the script is querying for an older block, then only archive node will return the result, which will cause a mismatch, however the mismatch doesn't mean archive node indexed invalidate data.

Reply:
addressed, we'll ignore this case for now.